client/components/select-dropdown/index.jsx

```diff
@@ -207,12 +207,20 @@ class SelectDropdown extends Component {
 			} );
 		}
 
-		let dropdownClassName = classNames( dropdownClasses );
-		let selectedText = this.props.selectedText
-			? this.props.selectedText
-			: result( find(
-				this.props.options, { value: this.state.selected }
+		const dropdownClassName = classNames( dropdownClasses );
+
+		let selectedText = this.props.selectedText;
```

Comment:
Code looks good here, but we can maybe decompose this a bit with a helper method:
```javascript
getSelectedText() {
    const { options, selectedText } = this.props;
    const { selected } = this.state;
    if ( selectedText ) {
        return selectedText;
    }
    //return currently selected text
    const selectedValue = selected ? selected : this.getInitialSelectedItem( this.props );
    return result( find( options, { value: selectedValue } ) );
}
```